### PR TITLE
v4

### PR DESCRIPTION
**Pull request by haf**
/cc @polytypic

Currently tryCommunicate never returns. Insights appreciated.

Related issue is here https://github.com/relentless/Http.fs/issues/94

### CODE REVIEW

File: HttpFs/HttpFs.fs [hunk 1]
```diff
@@ -863,17 +909,135 @@ module Client =
...
+        failwith "using CacheEntry with HttpFs not supported"
+
+      | other ->
+        failwith "There are only 21 options in this enum, something the F# compiler should be aware of"
+
+    let tryCommunicate (request : WebRequest) (xJ : Job<_>) fWebExnResp : Alt<Choice<'a, Error>> =
+      Alt.withNackJob <| fun nack ->
+        let rI = IVar ()
+
+        let rJ =
```
Comment (polytypic) [inline HttpFs/HttpFs.fs]: Is `rJ` an unused variable? I assume you want to start a new job here to write the result to `rI`.

**Comment by neoeinstein** [comment]
This PR seems to have a lot more than just the conversion to `Alt`. Is there a way to split up the changes some? I see that there are changes for `Alt`, logging, changing the signature of `getResponse`, and adding a failure DU. I think each of these could have their own PR.

**Comment by haf** [comment]
@neoeinstein Yes, this is me in "let's hack some fun code" mode. This PR contains a lot of breaking changes. I'm not done with it yet though – I'm trying to get to a stage where I can pipeline requests and create `Http.fs.ZipKin` that depends on `Logary.Zipkin` and properly adds tracing information to requests.

**Comment by haf** [comment]
@relentless Please make me admin or put releases/v3.x as the 'main branch'. Thanks 👍

**Comment by relentless** [comment]
I've transferred the repo to you, @haf.  You've done more with it than I ever planned to, and I doubt I'll be doing much more with it in the future.  Enjoy!

**Comment by haf** [comment]
Thanks @relentless, I appreciate it.

**Pull request closed without merging**